src/backend/env/mod.rs

```diff
@@ -1554,19 +1561,36 @@ impl State {
             )
         });
 
+        let log_time = |state: &mut State, frequency| {
+            state.logger.debug(format!(
+                "{} routine finished after `{}` ms.",
```

Comment:
FYI: this works on granularity of rounds (if there are async calls before). Otherwise the result will be 0.

Reply:
You're saying this is not the wall time, right? I expect that. I would be still curious to see what we get. Because I have no explanation for this: https://6qfxa-ryaaa-aaaai-qbhsq-cai.icp0.io/#/post/717170

Reply:
Replied on that thread. The change sound good for that purpose.

Reply:
I will just set it to current time as you suggested. The reason I add 1h is to not have it drifting...

Reply:
Ok, maybe in the next release.